src/Shared/UnitTests/FileMatcher_Tests.cs

```diff
@@ -995,8 +995,14 @@ private void GetMatchingDirectories(string[] candidates, string path, string pat
                             int nextSlash = normalizedCandidate.IndexOfAny(FileMatcher.directorySeparatorCharacters, path.Length + 1);
                             if (nextSlash != -1)
                             {
-                                string match = normalizedCandidate.Substring(0, nextSlash + 1);
+                                string match;
+
+                                match = normalizedCandidate.StartsWith(@"\\") && string.IsNullOrEmpty(path)
+                                    ? normalizedCandidate.Substring(0, nextSlash + 1)
+                                    : normalizedCandidate.Substring(0, nextSlash);
```

Comment:
After I made the test's mock file system not append slashes (to get it consistent with stuff in `System.IO` which does not do that when enumerating directories) a bunch of UNC tests started crashing.  UNC path need that +1 so the proper `\\` fragment is retained. I'll add a comment about it.